make/python_env.mk

```diff
@@ -0,0 +1,342 @@
+# - Linux
+#
+# Frequently used make command:
+#
+# - make up
+# - make clean
+# - make install
+# - make test
+# - make build_doc
+# - make view_doc
+# - make deploy_doc
+# - make reformat
+
+
+#--- User Defined Variable ---
+PACKAGE_NAME="login_analytics"
+
+# Python version Used for Development
+PY_VER_MAJOR="3"
+PY_VER_MINOR="6"
+PY_VER_MICRO="2"
+
+# If you use pyenv-virtualenv, set to "Y"
+USE_PYENV="N"
+
+# S3 Bucket Name
+DOC_HOST_BUCKET_NAME="login-gov-doc"
+
+
+#--- Derive Other Variable ---
+
+# Virtualenv Name
+VENV_NAME="${PACKAGE_NAME}_venv"
+
+# Project Root Directory
+GIT_ROOT_DIR=${shell git rev-parse --show-toplevel}
+PROJECT_ROOT_DIR=${shell pwd}
+
+ifeq (${OS}, Windows_NT)
+    DETECTED_OS := Windows
+else
+    DETECTED_OS := $(shell uname -s)
+endif
+
+# Windows
+ifeq (${DETECTED_OS}, Windows)
+    USE_PYENV="N"
+
+    VENV_DIR_REAL="${PROJECT_ROOT_DIR}/${VENV_NAME}"
+    BIN_DIR="${VENV_DIR_REAL}/Scripts"
+    SITE_PACKAGES="${VENV_DIR_REAL}/Lib/site-packages"
+    SITE_PACKAGES64="${VENV_DIR_REAL}/Lib64/site-packages"
+
+    GLOBAL_PYTHON="/c/Python${PY_VER_MAJOR}${PY_VER_MINOR}/python.exe"
+    OPEN_COMMAND="start"
+endif
+
+
+# MacOS
+ifeq (${DETECTED_OS}, Darwin)
+
+ifeq ($(USE_PYENV), "Y")
+    VENV_DIR_REAL="${HOME}/.pyenv/versions/${PY_VERSION}/envs/${VENV_NAME}"
+    VENV_DIR_LINK="${HOME}/.pyenv/versions/${VENV_NAME}"
+    BIN_DIR="${VENV_DIR_REAL}/bin"
+    SITE_PACKAGES="${VENV_DIR_REAL}/lib/python${PY_VER_MAJOR}.${PY_VER_MINOR}/site-packages"
+    SITE_PACKAGES64="${VENV_DIR_REAL}/lib64/python${PY_VER_MAJOR}.${PY_VER_MINOR}/site-packages"
+else
+    VENV_DIR_REAL="${PROJECT_ROOT_DIR}/${VENV_NAME}"
+    VENV_DIR_LINK="./${VENV_NAME}"
+    BIN_DIR="${VENV_DIR_REAL}/bin"
+    SITE_PACKAGES="${VENV_DIR_REAL}/lib/python${PY_VER_MAJOR}.${PY_VER_MINOR}/site-packages"
+    SITE_PACKAGES64="${VENV_DIR_REAL}/lib64/python${PY_VER_MAJOR}.${PY_VER_MINOR}/site-packages"
+endif
+
+    GLOBAL_PYTHON="python${PY_VER_MAJOR}.${PY_VER_MINOR}"
+    OPEN_COMMAND="open"
+endif
+
+
+# Linux
+ifeq (${DETECTED_OS}, Linux)
+    USE_PYENV="N"
+
+    VENV_DIR_REAL="${PROJECT_ROOT_DIR}/${VENV_NAME}"
+    VENV_DIR_LINK="${PROJECT_ROOT_DIR}/${VENV_NAME}"
+    BIN_DIR="${VENV_DIR_REAL}/bin"
+    SITE_PACKAGES="${VENV_DIR_REAL}/lib/python${PY_VER_MAJOR}.${PY_VER_MINOR}/site-packages"
+    SITE_PACKAGES64="${VENV_DIR_REAL}/lib64/python${PY_VER_MAJOR}.${PY_VER_MINOR}/site-packages"
+
+    GLOBAL_PYTHON="python${PY_VER_MAJOR}.${PY_VER_MINOR}"
+    OPEN_COMMAND="open"
+endif
+
+
+BASH_PROFILE_FILE="${HOME}/.bash_profile"
+
+BIN_ACTIVATE="${BIN_DIR}/activate"
+BIN_PYTHON="${BIN_DIR}/python"
+BIN_PIP="${BIN_DIR}/pip"
+BIN_PYTEST="${BIN_DIR}/pytest"
+BIN_SPHINX_START="${BIN_DIR}/sphinx-quickstart"
+BIN_TWINE="${BIN_DIR}/twine"
+BIN_TOX="${BIN_DIR}/tox"
+BIN_JUPYTER="${BIN_DIR}/jupyter"
+
+
+S3_PREFIX="s3://${DOC_HOST_BUCKET_NAME}/${PACKAGE_NAME}"
+RTD_DOC_URL="https://login_analytics.readthedocs.io/index.html"
```

Comment:
Will this automatically publish the docs on readthedocs.io? If so, I don't think we want that. We want the docs to remain internal.

Reply:
We don't, we use s3 to host it, https://s3-us-west-2.amazonaws.com/login-gov-doc/login_analytics/index.html. Every time we just do `make build_doc`, `make deploy_doc`

Brian has some setup to restrict access only from GSA IP address.